src/index.tsx

```diff
@@ -1751,13 +1756,27 @@ export class App extends React.Component<any, AppState> {
                 return;
               }
               const hasDeselectedButton = Boolean(e.buttons);
+
+              const { x, y } = viewportCoordsToSceneCoords(e, this.state);
+              if (this.state.multiElement) {
+                const originX = this.state.multiElement.x;
+                const originY = this.state.multiElement.y;
+                const points = this.state.multiElement.points;
+                const pnt = points[points.length - 1];
+                pnt[0] = x - originX;
+                pnt[1] = y - originY;
+                this.state.multiElement.shape = null;
```

Comment:
should we fix this via:
```js
this.setState({
  multiElement: {
    ...this.state.multiElement,
    shape: null,
  },
});
```
or add ignore flag, for perf reasons?

Reply:
Mhm.. that proposed fix doesn't even work :thinking:

Reply:
I am not certain that copying the object here is going to work because we are mutating an object located in the elements array. This is also how we are setting shape to null for `draggingElement` and others. 

If you remember, when I got into this project, one of the things that I wanted to do was to make all interactions immutable. Over time, as I worked on multi arrows, I have kind of built up the following opinion regarding this project -- elements array should be immutable because it is the gateway between canvas and React; however, objects within those elements should not be immutable because it would take a lot of processing power to find, copy, mutate, and put it back to the array. This opinion might be weird or completely wrong  and I am open for suggestions. However, so far this is the path I took when working on multi-arrows.

Weirdly, I am not sure why this is giving an error because there are no errors when dragging element etc set values directly. Maybe, writing something like this fixes the issue:

```js
const multiElement = this.state.multiElement;
multiElement.shape = null;
```j

Reply:
Yes, eslint catches that only because you're setting it directly via `this.state.multiElement = `. So we can fix it either way (either disable the rule for that line, or destructure the variable).

Reply:
And I agree. Making members of the `elements` array mutable is desirable due to perf.

I'm still not sure why resetting the `shape` by spreading in `setState` doesn't work .

Reply:
> I'm still not sure why resetting the shape by spreading in setState doesn't work .
 
It is because you are creating a new object and setting it into state. As a result, you are mutating an object that is not in the elements array.

Reply:
Oh.. right, I forgot that `multiElement` is just a reference to an element from `elements` array. Yea, that makes sense :man_facepalming:.